### PR TITLE
issue #324: base information form

### PR DESCRIPTION
**Pull request by k-allagbe**
- [x] input component for weight, density & volume
- [x] data types
- [x] base information form component
- [x] integration with stepper
- [x] tests

### CODE REVIEW

**Comment by SamuelPelletierEvraire** [comment]
For better grouping of the translation file, I think we should use something like this :
public/locales/en/FormPage.json

```
FormPage.json:
baseInformation:{
    key1:value1,
    key2:value2,
    ...
},
labelDataValidation:{
    key1:value1,
    key2:value2,
    ...
},
VerifiedQuantityMultyInput:{
    key1:value1,
    key2:value2,
     "units": {
     
     },
    "error":{
    
     },
     ...
}
```

Please see https://github.com/ai-cfia/fertiscan-frontend/pull/332 to see how the nomenclature was done and approved.

**Review by SamuelPelletierEvraire — CHANGES_REQUESTED** [review]
Gonna check the rest when the translation is made don't want to add to much request of change. 
Please don't forget that each icon/Image need alt or arial-label (on parent of the component for arial-label) for accessibility approval.

**Comment by k-allagbe** [comment]
> For better grouping of the translation file, I think we should use something like this : public/locales/en/FormPage.json
> 
> ```
> FormPage.json:
> baseInformation:{
>     key1:value1,
>     key2:value2,
>     ...
> },
> labelDataValidation:{
>     key1:value1,
>     key2:value2,
>     ...
> },
> VerifiedQuantityMultyInput:{
>     key1:value1,
>     key2:value2,
>      "units": {
>      
>      },
>     "error":{
>     
>      },
>      ...
> }
> ```
> 
> Please see #332 to see how the nomenclature was done and approved.

done

File: src/components/ImageViewer.tsx [hunk 1]
```diff
@@ -131,55 +131,68 @@ const ControlBar: React.FC<ControlBarProps> = ({
...
-          onClick={() => swiper?.slideNext()}
-          data-testid="next-button"
-          disabled={activeIndex >= zoomRefs.length - 1}
-        >
-          <ArrowForwardIosIcon />
-        </Button>
+      <Tooltip
+        title="Next"
+        disableHoverListener={activeIndex >= zoomRefs.length - 1}
+      >
```
[the remark this answers is not in the document]
Comment (k-allagbe) [reply src/components/ImageViewer.tsx:149]: Will do in another PR.

File: src/components/OrganizationsForm.tsx [hunk 2]
```diff
@@ -96,19 +82,17 @@ const OrganizationsForm: React.FC<FormComponentProps> = ({
...
                   title="Mark all as Verified"
                   enterDelay={1000}
-                  disableHoverListener={areAllFieldStatus(
-                    index,
-                    FieldStatus.Verified,
+                  disableHoverListener={isVerified(
+                    watchedOrganizations?.[index],
+                    true,
                   )}
                 >
```
[the remark this answers is not in the document]
Comment (k-allagbe) [reply src/components/OrganizationsForm.tsx:89]: Will do in another PR.

File: src/components/OrganizationsForm.tsx [hunk 3]
```diff
@@ -96,19 +82,17 @@ const OrganizationsForm: React.FC<FormComponentProps> = ({
...
                       color="secondary"
-                      onClick={() =>
-                        setAllFieldsStatus(index, FieldStatus.Verified)
-                      }
-                      disabled={areAllFieldStatus(index, FieldStatus.Verified)}
+                      onClick={() => setAllVerified(index, true)}
+                      disabled={isVerified(watchedOrganizations?.[index], true)}
                       data-testid={`verify-all-btn-${index}`}
                     >
                       <DoneAllIcon />
```
[the remark this answers is not in the document]
Comment (k-allagbe) [reply src/components/OrganizationsForm.tsx:98]: Will do in another PR.

File: src/components/OrganizationsForm.tsx [hunk 4]
```diff
@@ -118,21 +102,19 @@ const OrganizationsForm: React.FC<FormComponentProps> = ({
                 <Tooltip
                   title="Mark all as Unverified"
                   enterDelay={1000}
```
[the remark this answers is not in the document]
Comment (k-allagbe) [reply src/components/OrganizationsForm.tsx:104]: Will do in another PR.

File: src/components/OrganizationsForm.tsx [hunk 5]
```diff
@@ -148,7 +130,7 @@ const OrganizationsForm: React.FC<FormComponentProps> = ({
                       onClick={() => remove(index)}
                       data-testid={`remove-org-btn-${index}`}
                     >
-                      <RemoveIcon />
+                      <DeleteIcon />
                     </Button>
                   </span>
```
[the remark this answers is not in the document]
Comment (k-allagbe) [reply src/components/OrganizationsForm.tsx:135]: Will do in another PR.

File: src/components/ImageViewer.tsx [hunk 6]
```diff
@@ -131,55 +131,68 @@ const ControlBar: React.FC<ControlBarProps> = ({
...
+      <Tooltip title="Zoom In" disableHoverListener={!currentZoomRef}>
+        <span>
+          <Button
+            variant="contained"
+            onClick={() => currentZoomRef?.zoomIn()}
+            data-testid="zoom-in-button"
+            disabled={!currentZoomRef}
+          >
+            <ZoomInIcon />
+          </Button>
```
[the remark this answers is not in the document]
Comment (k-allagbe) [reply src/components/ImageViewer.tsx:170]: Will do in another PR.

File: src/components/ImageViewer.tsx [hunk 7]
```diff
@@ -131,55 +131,68 @@ const ControlBar: React.FC<ControlBarProps> = ({
...
+      <Tooltip title="Zoom Out" disableHoverListener={!currentZoomRef}>
+        <span>
+          <Button
+            variant="contained"
+            onClick={() => currentZoomRef?.zoomOut()}
+            data-testid="zoom-out-button"
+            disabled={!currentZoomRef}
+          >
+            <ZoomOutIcon />
+          </Button>
```
[the remark this answers is not in the document]
Comment (k-allagbe) [reply src/components/ImageViewer.tsx:182]: Will do in another PR.

File: src/components/ImageViewer.tsx [hunk 8]
```diff
@@ -131,55 +131,68 @@ const ControlBar: React.FC<ControlBarProps> = ({
...
+        <span>
+          <Button
+            variant="contained"
+            onClick={() => zoomRefs.forEach((ref) => ref?.resetTransform())}
+            data-testid="reset-button"
+            disabled={!currentZoomRef}
+          >
+            <YoutubeSearchedForIcon />
+          </Button>
+        </span>
```
[the remark this answers is not in the document]
Comment (k-allagbe) [reply src/components/ImageViewer.tsx:195]: Will do in another PR.

**Review by ChromaticPanic — APPROVED** [review]

**Pull request merged**